src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -667,7 +667,8 @@ private void WriteNonceCookie(string nonce)
                 new CookieOptions
                 {
                     HttpOnly = true,
-                    Secure = Request.IsHttps
+                    Secure = Request.IsHttps,
+                    Expires = DateTime.UtcNow + Options.ProtocolValidator.NonceLifetime
```

Comment:
CookieOptions.Expires might just be an oversight. I don't see a reason. File a bug :-)

We could use ISystemClock, but I wouldn't bother for just one test. Interestingly JwtBearerAuthenticationOptions does have an ISystemClock option but does not actually use it.